__tests__/p2sh/tx-proposal-create.test.js

```diff
@@ -151,4 +151,31 @@ describe('create tx-proposal api', () => {
     expect(response.status).toBe(200);
     expect(response.body.success).toBe(false);
   });
+
+  it('should mark utxos as used when sending mark_as_used', async () => {
+    const selectSpy = jest.spyOn(SendTransaction.prototype, 'updateOutputSelected').mockImplementationOnce(jest.fn(async () => {}));
+    try {
+      const response = await TestUtils.request
+        .post('/wallet/p2sh/tx-proposal')
+        .send({
+          outputs: [
+            { address: 'WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc', value: 1 },
+            { address: 'wcUZ6J7t2B1s8bqRYiyuZAftcdCGRSiiau', value: 1 },
+          ],
+          mark_inputs_as_used: true,
+        })
+        .set({ 'x-wallet-id': walletId });
+      expect(response.status).toBe(200);
+      console.log(JSON.stringify(response.body));
+      expect(response.body.txHex).toBeDefined();
+      expect(response.body.success).toBe(true);
+      const tx = hathorLib.helpersUtils.createTxFromHex(response.body.txHex, new hathorLib.Network('testnet'));
+      expect(tx.outputs.map(o => o.decodedScript.address.base58))
+        .toEqual(expect.arrayContaining(['WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc', 'wcUZ6J7t2B1s8bqRYiyuZAftcdCGRSiiau']));
+
+      expect(selectSpy).toHaveBeenCalledTimes(1);
```

Comment:
[question] Should we extend the expect tests to check also the call arguments?

Reply:
We can but in this specific test we dont know the input being selected.
It should select one from the history fixture and if we just copy the details and the fixture changes in the future, the test would fail without explanation.